python/dgl/graph.py

```diff
@@ -3451,6 +3455,24 @@ def local_scope(self):
         old_eframe = self._edge_frame
         self._node_frame = FrameRef(Frame(self._node_frame._frame))
         self._edge_frame = FrameRef(Frame(self._edge_frame._frame))
+        sync_frame_initializer(self._node_frame._frame, old_nframe._frame)
+        sync_frame_initializer(self._edge_frame._frame, old_eframe._frame)
         yield
         self._node_frame = old_nframe
         self._edge_frame = old_eframe
+
+def sync_frame_initializer(new_frame, reference_frame):
```

Comment:
I recommend moving this function to frame.py

Reply:
Done.